sim/battle.ts

```diff
@@ -103,8 +105,9 @@ export class Battle extends Dex.ModdedDex {
 		this.inputLog = [];
 		this.sentLogPos = 0;
 		this.sentEnd = false;
+		this.gameType = (format.gameType || 'singles');
 		// @ts-ignore
-		this.sides = [null, null];
+		this.sides = this.gameType === 'multi' || this.gameType === 'free-for-all' ? [null, null, null, null] : [null, null];
```

Comment:
Apparently `Array.fill` exists.

```
const isFourPlayer = this.gameType === 'multi' || this.gameType === 'free-for-all';
this.sides = Array(isFourPlayer ? 4 : 2).fill(null!);
```

Reply:
Why doesn't TypeScript complain about this?

Reply:
`null!` instead of `null`.

A lot of old `@ts-ignore`s were just be because I don't have access to the `!` non-null assertion operator.